### PR TITLE
Dependency updates & OpenSSL/unbound removal

### PR DESCRIPTION
**Pull request by jagerman**
- Updates various static and submodule dependencies.
- Switch our sha256 implementation to use libsodium instead of openssl.
- Transitions from oxen-mq to oxen-encoding library for bt/hex/b64/etc. encoding duties (since we extracted all the encoding bits into its own header-only library).
- Clean out some epee garbage
- Remove some moronic use of wipeable_string from wallet2 where the wipeable_string is pointless because the value is immediately copied into a regular string.
- ~~cleaned up some macro vomit~~ (Dupe of #1516)
- Removed miniupnpc entirely (it's been disabled for more than a year).
- Address a compiler warning.

## Edit: OpenSSL & unbound removal

OpenSSL was being a pain in the ass on iOS, so I just removed it from all our dependencies:
- it gets (mis)used to write base64 data to a file in wallet2 for things like multisig data.  However this mode is *not* enabled by default, basically completely unknown, completely unused, only exists in the cli wallet, and is just dumb.  (Honestly the justification given in the Monero PR that added it is that "Windows users might want it", presupposing that there exists Windows users who are capable of generating a multisig wallet in a CLI-only application and yet are incapable of dealing with binary files).

- it's a dependency of unbound (in order to do dnssec, I believe).  Unbound itself is fairly useless for Oxen, so I've removed it too:
    - it does OpenAlias lookups, which are a Monero thing that has never been used outside Monero, doesn't work reliably (because it fails if the result isn't DNSSEC validated) and is pointless when we have ONS.

    - it does DNS resolution on seed nodes, but we have never set seed nodes by name and just use seed node IPs instead (which seems a bit better anyway since the DNS lookup leaks some metadata).

- it *was* being used for sha256, but an earlier commit in this PR already replaced that with libsodium (entirely coincidentally).

- for static deps, it enables HTTPS support for the wallet.  However only the CLI wallet actually supports this (the GUI and mobile wallets don't), and since oxend hasn't support this for a while I have strong doubts it is being used anywhere.  (Plus wallet3 will do everything encrypted using zmq/libsodium, so doesn't need this to be secure).  Note here that it is *only* removed by this commit for static builds: if doing a system build that links to libcurl supporting HTTPS then HTTPS support will still work.

Libexpat is also gone because it was only there for libunbound.

### CODE REVIEW

File: src/simplewallet/simplewallet.cpp [hunk 1]
```diff
@@ -6714,7 +6645,7 @@ bool simple_wallet::ons_update_mapping(std::vector<std::string> args)
     }
 
     auto& enc_hex = response[0].encrypted_value;
-    if (!oxenmq::is_hex(enc_hex) || enc_hex.size() % 2 != 0 || enc_hex.size() > 2*ons::mapping_value::BUFFER_SIZE)
+    if (!oxenc::is_hex(enc_hex) || enc_hex.size() % 2 != 0 || enc_hex.size() > 2*ons::mapping_value::BUFFER_SIZE)
```
Comment (tewinget) [inline src/simplewallet/simplewallet.cpp:6648]: isn't the size mod 0 check redundant with the one preceding?

File: src/simplewallet/simplewallet.cpp [hunk 2]
```diff
@@ -6956,7 +6887,7 @@ bool simple_wallet::ons_lookup(std::vector<std::string> args)
   for (auto const &mapping : response)
   {
     auto& enc_hex = mapping.encrypted_value;
-    if (mapping.entry_index >= args.size() || !oxenmq::is_hex(enc_hex) || enc_hex.size() % 2 != 0 || enc_hex.size() > 2*ons::mapping_value::BUFFER_SIZE)
+    if (mapping.entry_index >= args.size() || !oxenc::is_hex(enc_hex) || enc_hex.size() % 2 != 0 || enc_hex.size() > 2*ons::mapping_value::BUFFER_SIZE)
```
Comment (tewinget) [inline src/simplewallet/simplewallet.cpp:6890]: same as above

File: src/simplewallet/simplewallet.cpp (hunk 1 above)
Comment (jagerman) [reply src/simplewallet/simplewallet.cpp:6648]: Yes, but in some early versions of oxenmq `is_hex` it wasn't which it why it was here.  It is now, though, so I'll update.

**Review by tewinget — APPROVED** [review]
Seems fine, pending CI.

**Pull request merged**